go/libraries/doltcore/sqle/dsess/mutexmap/mutexmap.go

```diff
@@ -0,0 +1,82 @@
+// Source: https://stackoverflow.com/questions/40931373/how-to-gc-a-map-of-mutexes-in-go
```

Comment:
Probably just want to use Dolthub copyright here

Or at least include the Doltub copyright notice as well, before other attributions

Reply:
Replaced with my own implementation.